tsl/src/bgw_policy/continuous_aggregate_api.c

```diff
@@ -132,6 +133,80 @@ policy_refresh_cagg_get_refresh_end(const Dimension *dim, const Jsonb *config)
 	return res;
 }
 
+/* returns false if a policy could not be found */
+bool
+policy_refresh_cagg_exists(int32 materialization_id)
+{
+	Hypertable *mat_ht = ts_hypertable_get_by_id(materialization_id);
+
+	if (!mat_ht)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("configuration materialization hypertable id %d not found",
+						materialization_id)));
+
+	List *jobs = ts_bgw_job_find_by_proc_and_hypertable_id(POLICY_REFRESH_CAGG_PROC_NAME,
+														   INTERNAL_SCHEMA_NAME,
+														   materialization_id);
+	if (jobs == NIL)
+		return false;
+
+	/* only 1 cont. aggregate policy job allowed */
+	Assert(list_length(jobs) == 1);
+	return true;
+}
+
+/* Compare passed in interval value with refresh_start parameter
+ * of cagg policy.
+ */
+bool
+policy_refresh_cagg_refresh_start_lt(int32 materialization_id, Oid cmp_type, Datum cmp_interval)
+{
+	Hypertable *mat_ht = ts_hypertable_get_by_id(materialization_id);
+	bool ret = false;
+
+	if (!mat_ht)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("configuration materialization hypertable id %d not found",
+						materialization_id)));
+
+	List *jobs = ts_bgw_job_find_by_proc_and_hypertable_id(POLICY_REFRESH_CAGG_PROC_NAME,
+														   INTERNAL_SCHEMA_NAME,
+														   materialization_id);
+	if (jobs == NIL)
+		return false;
```

Comment:
The code path needs a corresponding test case, if it's possible to add one. Otherwise, please add a comment that it should never happen in practice and is needed only for debugging purposes.